.travis.yml

```diff
@@ -1,17 +1,47 @@
 language: python
 sudo: false
-python:
-    - 2.6
-    - 2.7
-    - 3.2
-    - 3.3
-    - 3.4
-    - 3.5
-    - pypy
-    - pypy3
+matrix:
+    include:
+        - os: linux
+          python: 2.6
+        - os: linux
+          python: 2.7
+        - os: linux
+          python: 3.3
+        - os: linux
+          python: 3.4
+        - os: linux
+          python: 3.5
+        - os: linux
+          python: pypy
+        - os: linux
+          python: pypy3
+        - os: osx
+          language: generic
+          env: TERRYFY_PYTHON='homebrew 2'
+        - os: osx
+          language: generic
+          env: TERRYFY_PYTHON='homebrew 3'
+before_install:
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then git clone https://github.com/MacPython/terryfy; fi
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then source terryfy/travis_tools.sh; fi
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then get_python_environment $TERRYFY_PYTHON venv; fi
+    - if [[ "$TERRYFY_PYTHON" == "homebrew 3" ]]; then alias pip=`which pip3` ; fi
 install:
     - pip install .
 script:
     - python setup.py test -q
 notifications:
     email: false
+after_success:
+    - echo [distutils]                                  > ~/.pypirc
+    - echo index-servers = pypi                        >> ~/.pypirc
+    - echo [pypi]                                      >> ~/.pypirc
+    - echo repository=https://pypi.python.org/pypi     >> ~/.pypirc
+    - echo username=USER                               >> ~/.pypirc
+    - echo password=$PYPIPASSWORD                      >> ~/.pypirc
+    - if [[ $TRAVIS_TAG && "$TRAVIS_OS_NAME" == "osx" ]]; then python setup.py bdist_wheel upload; fi
+    - if [[ $TRAVIS_TAG && "$TRAVIS_OS_NAME" == "linux" ]]; then python setup.py sdist upload; fi
```

Comment:
I don't think we need this: the usual release process is "somebody runs [zest.releaser](https://pypi.python.org/pypi/zest.releaser)'s `fullrelease` script", which

1. updates CHANGELOG.rst and version number (removing the .dev0 suffix) in setup.py
2. commits and tags locally
3. builds an sdist and uploads it to PyPI
4. creates new CHANGELOG.rst section and bumps version number (and adds .dev0) in setup.py
5. pushes new commits and tags to GitHub

so at this point I expect an sdist for this tag to already exist.  Won't trying to upload a new one result in an error and a failed build?